src/System.Windows.Forms/tests/UnitTests/misc/CollectionHelperTests.cs

```diff
@@ -0,0 +1,149 @@
+        {
+            var source = new Dictionary<string, string>()
+            {
+                { "key-one", "value-one" },
+                { "key-two", "value-two" },
+            };
+            var target = new object[3];
+
+            Assert.Throws<ArgumentException>(() => source.HashtableCopyTo(target, 2));
+        }
+
+        [Fact]
+        public void HashtableCopyTo_throws_when_target_is_null_object_array()
+        {
+            var source = new Dictionary<string, string>();
+            var target = (object[])null;
+            var index = 0;
+
+            Assert.Throws<ArgumentNullException>(() => source.HashtableCopyTo(target, index));
+        }
+
+        [Fact]
+        public void HashtableCopyTo_successfully_copies_to_KeyValuePair_array()
+        {
+            var source = new Dictionary<string, string>()
+            {
+                { "key-one", "value-one" },
+                { "key-two", "value-two" },
+            };
+            var target = new KeyValuePair<string, string>[2];
+
+            source.HashtableCopyTo(target, 0);
+
+            var firstTargetItem = target[0];
+            Assert.Equal(typeof(KeyValuePair<string, string>), firstTargetItem.GetType());
+            Assert.Equal("key-one", firstTargetItem.Key);
+            Assert.Equal("value-one", firstTargetItem.Value);
+
+            var secondTargetItem = target[1];
+            Assert.Equal(typeof(KeyValuePair<string, string>), secondTargetItem.GetType());
+            Assert.Equal("key-two", secondTargetItem.Key);
+            Assert.Equal("value-two", secondTargetItem.Value);
+        }
+
+        [Fact]
+        public void HashtableCopyTo_successfully_copies_to_DictionaryEntry_array()
+        {
+            var source = new Dictionary<string, string>()
+            {
+                { "key-one", "value-one" },
+                { "key-two", "value-two" },
+            };
+            var target = new DictionaryEntry[2];
+
+            source.HashtableCopyTo(target, 0);
+
+            var firstTargetItem = target[0];
+            Assert.Equal(typeof(DictionaryEntry), firstTargetItem.GetType());
+            Assert.Equal("key-one", (string)firstTargetItem.Key);
+            Assert.Equal("value-one", (string)firstTargetItem.Value);
+
+            var secondTargetItem = target[1];
+            Assert.Equal(typeof(DictionaryEntry), secondTargetItem.GetType());
+            Assert.Equal("key-two", (string)secondTargetItem.Key);
+            Assert.Equal("value-two", (string)secondTargetItem.Value);
+        }
+
+        [Fact]
+        public void HashtableCopyTo_successfully_copies_to_Object_array()
+        {
+            var source = new Dictionary<string, string>()
+            {
+                { "key-one", "value-one" },
+                { "key-two", "value-two" },
+            };
+            var target = new DictionaryEntry[2];
+
+            source.HashtableCopyTo(target, 0);
```

Comment:
```suggestion
            source.HashtableCopyTo(target, index: 0);
```
constants should always carry param names for readability.